### ISSUE TITLE
Restart kubelet causes the virt-laucher pod terminated

### ISSUE THREAD

**Issue by caohuilong** [kubevirt/kubevirt#12290]
## **What happened**:
When I restart the kubelet, **the virt-launcher pods in the node terminated** and the other pod will not be affected. 
We have many k8s clusters, this problem can be reproduced 100% in one of cluster and will not appeared in the other clusters.

## **Some Infos**
After the kubelet restarted, the event list is like this:
![image](https://github.com/kubevirt/kubevirt/assets/26712324/a8a7e24c-4ed3-4fed-aca9-ad9b196f59fb)
The log file of virt-handler and virt-launcher.
[virt-handler-942th.log](https://github.com/user-attachments/files/16097002/virt-handler-942th.log)
[virt-launcher-debian-dv-jh7q4.log](https://github.com/user-attachments/files/16097004/virt-launcher-debian-dv-jh7q4.log)

By analyzing the logs, i found that it's **the virt-launcher pod firstly recive the SIGTERM signal**, but i cannot tell where the SIGTERM is from.

## **What you expected to happen**:
I hope that restart kubelet will not affect the virt-laucher pod.

## **How to reproduce it (as minimally and precisely as possible)**:
run `systemctl restart kubelet` in the node.

## **Additional context**:
I have noticed this issue [Virt-handler heartbeat causes the vm to be restarted when kubelet is restarted.](https://github.com/kubevirt/kubevirt/issues/11843), but i schedule a normal pod with nodeSelector: kubevirt.io/schedulable=true，the pod will not terminate when restarting the kubelet. So i think it's not the same problem.

## **Environment**:
- KubeVirt version (use `virtctl version`): v0.36.0
- Kubernetes version (use `kubectl version`): v1.15.5
- VM or VMI specifications: N/A
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): Debian GNU/Linux 9 (stretch)
- Kernel (e.g. `uname -a`): 4.9.0-13-amd64
- Container Runtime Version: docker://19.3.12

**Comment by Homura222** [comment]
i think you can see https://github.com/kubevirt/kubevirt/issues/11843#issuecomment-2093232032 and https://github.com/kubevirt/kubevirt/issues/11843#issuecomment-2097600636

`After kubelet restarts, kubelet will kill pods that do not match the node label.（virt-launcher pod nodeSelector is kubevirt.io/schedulable: "true"）At this time, the node kubevirt.io/schedulable label value is false.The virt-launcher pod will be killed by kubelet, causing the vm to restart.`

**Comment by caohuilong** [comment]
> I think you can see [#11843 (comment)](https://github.com/kubevirt/kubevirt/issues/11843#issuecomment-2093232032) and [#11843 (comment)](https://github.com/kubevirt/kubevirt/issues/11843#issuecomment-2097600636)
> 
> `After kubelet restarts, kubelet will kill pods that do not match the node label.（virt-launcher pod nodeSelector is kubevirt.io/schedulable: "true"）At this time, the node kubevirt.io/schedulable label value is false.The virt-launcher pod will be killed by kubelet, causing the vm to restart.`
> 
> Have you checked the `kubelet` logs?

I have noticed this issue https://github.com/kubevirt/kubevirt/issues/11843, but i schedule a normal pod with nodeSelector: kubevirt.io/schedulable=true，the pod will not terminate when restarting the kubelet. So i think it's not the same problem.

**Comment by Homura222** [comment]
> > I think you can see [#11843 (comment)](https://github.com/kubevirt/kubevirt/issues/11843#issuecomment-2093232032) and [#11843 (comment)](https://github.com/kubevirt/kubevirt/issues/11843#issuecomment-2097600636)
> > `After kubelet restarts, kubelet will kill pods that do not match the node label.（virt-launcher pod nodeSelector is kubevirt.io/schedulable: "true"）At this time, the node kubevirt.io/schedulable label value is false.The virt-launcher pod will be killed by kubelet, causing the vm to restart.`
> > Have you checked the `kubelet` logs?
> 
> I have noticed this issue #11843, but i schedule a normal pod with nodeSelector: kubevirt.io/schedulable=true，the pod will not terminate when restarting the kubelet. So i think it's not the same problem.

Did you delete the `kubevirt.io/schedulable=true` label from the `node` before restarting kubelet?

**Comment by caohuilong** [comment]
> > > I think you can see [#11843 (comment)](https://github.com/kubevirt/kubevirt/issues/11843#issuecomment-2093232032) and [#11843 (comment)](https://github.com/kubevirt/kubevirt/issues/11843#issuecomment-2097600636)
> > > `After kubelet restarts, kubelet will kill pods that do not match the node label.（virt-launcher pod nodeSelector is kubevirt.io/schedulable: "true"）At this time, the node kubevirt.io/schedulable label value is false.The virt-launcher pod will be killed by kubelet, causing the vm to restart.`
> > > Have you checked the `kubelet` logs?
> > 
> > 
> > I have noticed this issue #11843, but i schedule a normal pod with nodeSelector: kubevirt.io/schedulable=true，the pod will not terminate when restarting the kubelet. So i think it's not the same problem.
> 
> Did you set the `node` `kubevirt.io/schedulable` label to `false` before restarting kubelet?

No, i didn't. I have tested that if the label of the node changed after the pod scheduled to the node, and the pod would become error after the kubelet restart, as you tested in the  [comment](https://github.com/kubevirt/kubevirt/issues/11843#issuecomment-2097600636).

**Comment by Homura222** [comment]
> > > > I think you can see [#11843 (comment)](https://github.com/kubevirt/kubevirt/issues/11843#issuecomment-2093232032) and [#11843 (comment)](https://github.com/kubevirt/kubevirt/issues/11843#issuecomment-2097600636)
> > > > `After kubelet restarts, kubelet will kill pods that do not match the node label.（virt-launcher pod nodeSelector is kubevirt.io/schedulable: "true"）At this time, the node kubevirt.io/schedulable label value is false.The virt-launcher pod will be killed by kubelet, causing the vm to restart.`
> > > > Have you checked the `kubelet` logs?
> > > 
> > > 
> > > I have noticed this issue #11843, but i schedule a normal pod with nodeSelector: kubevirt.io/schedulable=true，the pod will not terminate when restarting the kubelet. So i think it's not the same problem.
> > 
> > 
> > Did you set the `node` `kubevirt.io/schedulable` label to `false` before restarting kubelet?
> 
> No, i didn't. I have tested that if the label of the node changed after the pod scheduled to the node, and the pod would become error after the kubelet restart, as you tested in the [comment](https://github.com/kubevirt/kubevirt/issues/11843#issuecomment-2097600636).

```
3. wait pod running
4. kubectl label node ${ip} a-
5. systemctl restart kubelet
6. watch pod become error
```
I delete the `a` label before restarting kubelet. Then the pod affinity will not match the node label before restarting kubelet.

**Comment by caohuilong** [comment]
> > > > > I think you can see [#11843 (comment)](https://github.com/kubevirt/kubevirt/issues/11843#issuecomment-2093232032) and [#11843 (comment)](https://github.com/kubevirt/kubevirt/issues/11843#issuecomment-2097600636)
> > > > > `After kubelet restarts, kubelet will kill pods that do not match the node label.（virt-launcher pod nodeSelector is kubevirt.io/schedulable: "true"）At this time, the node kubevirt.io/schedulable label value is false.The virt-launcher pod will be killed by kubelet, causing the vm to restart.`
> > > > > Have you checked the `kubelet` logs?
> > > > 
> > > > 
> > > > I have noticed this issue #11843, but i schedule a normal pod with nodeSelector: kubevirt.io/schedulable=true，the pod will not terminate when restarting the kubelet. So i think it's not the same problem.
> > > 
> > > 
> > > Did you set the `node` `kubevirt.io/schedulable` label to `false` before restarting kubelet?
> > 
> > 
> > No, i didn't. I have tested that if the label of the node changed after the pod scheduled to the node, and the pod would become error after the kubelet restart, as you tested in the [comment](https://github.com/kubevirt/kubevirt/issues/11843#issuecomment-2097600636).
> 
> ```
> 3. wait pod running
> 4. kubectl label node ${ip} a-
> 5. systemctl restart kubelet
> 6. watch pod become error
> ```
> 
> I delete the node `a` label before restarting kubelet. Then the pod affinity will not match the node label before restarting kubelet.

加个vx聊一下可以吗？我的邮箱cao130507@163.com

**Comment by caohuilong** [comment]
> I think you can see [#11843 (comment)](https://github.com/kubevirt/kubevirt/issues/11843#issuecomment-2093232032) and [#11843 (comment)](https://github.com/kubevirt/kubevirt/issues/11843#issuecomment-2097600636)
> 
> `After kubelet restarts, kubelet will kill pods that do not match the node label.（virt-launcher pod nodeSelector is kubevirt.io/schedulable: "true"）At this time, the node kubevirt.io/schedulable label value is false.The virt-launcher pod will be killed by kubelet, causing the vm to restart.`
> 
> Have you checked the `kubelet` logs?

I have checked the kubelet but didn't found any clues, you can check the kubelet.log in this issue.

**Comment by galbeniluz** [comment]
Is there any workaround for this issue? this is blocking us from upgrading Kubernetes

**Comment by fabiand** [comment]
The workaround is to

1. cordon node
2. drain node
3. upgrade node
4. uncordon node

tldr; updat ethe node with VMs being elsewhere.

**Comment by xpivarc** [comment]
/assign
Are any others Pods using any device? If I remember correctly  this is general issue for any Pod using any device.

**Comment by maiqueb** [comment]
> The workaround is to
> 
>     1. cordon node
> 
>     2. drain node
> 
>     3. upgrade node
> 
>     4. uncordon node
> 
> 
> tldr; updat ethe node with VMs being elsewhere.

@galbeniluz does this workaround work for you ?
